src/shared/components/home/emojis-form.tsx

```diff
@@ -279,37 +298,80 @@ export class EmojiForm extends Component<EmojiFormProps, EmojiFormState> {
     );
   }
 
-  canEdit(cv: CustomEmojiViewForm) {
-    const noEmptyFields =
-      cv.alt_text.length > 0 &&
-      cv.category.length > 0 &&
-      cv.image_url.length > 0 &&
-      cv.shortcode.length > 0;
-    const noDuplicateShortCodes =
-      this.state.customEmojis.filter(
-        x => x.shortcode === cv.shortcode && x.id !== cv.id,
-      ).length === 0;
-    return noEmptyFields && noDuplicateShortCodes && !cv.loading && cv.changed;
+  canSave(cv: EditableEmoji) {
+    const requiredFields =
+      cv.emoji.custom_emoji.image_url.length > 0 &&
+      cv.emoji.custom_emoji.shortcode.length > 0;
+    return requiredFields && !cv.loading;
   }
 
-  getEditTooltip(cv: CustomEmojiViewForm) {
-    if (this.canEdit(cv)) return I18NextService.i18n.t("save");
+  getEditTooltip(cv: EditableEmoji) {
+    if (this.canSave(cv)) return I18NextService.i18n.t("save");
     else return I18NextService.i18n.t("custom_emoji_save_validation");
   }
 
-  handlePageChange(page: number) {
-    this.setState({ page: page });
+  async handlePageChange(page: number) {
+    this.setState({ loading: true });
+    let allEmojis: CustomEmojiView[] = this.state.allEmojis;
+    let emojiMartCustom: EmojiMartCategory[] = this.state.emojiMartCustom;
+    let emojiMartKey: number = this.state.emojiMartKey;
+    if (this.needsRefetch) {
+      const emojiRes = await HttpService.client.listCustomEmojis({
+        ignore_page_limits: true,
+      });
+      if (emojiRes.state === "success") {
+        this.needsRefetch = false;
+        allEmojis = emojiRes.data.custom_emojis;
+        allEmojis.sort((a, b) => {
+          if (a.custom_emoji.category < b.custom_emoji.category) {
+            return -1;
+          }
+          if (a.custom_emoji.category > b.custom_emoji.category) {
+            return 1;
+          }
+          if (a.custom_emoji.shortcode < b.custom_emoji.shortcode) {
+            return -1;
+          }
+          if (a.custom_emoji.shortcode > b.custom_emoji.shortcode) {
+            return 1;
+          }
```

Comment:
Is it viable to lump these together? e.g.
```suggestion
          if (a.custom_emoji.category < b.custom_emoji.category || a.custom_emoji.shortcode < b.custom_emoji.shortcode) {
            return -1;
          }
          if (a.custom_emoji.category > b.custom_emoji.category || a.custom_emoji.shortcode > b.custom_emoji.shortcode) {
            return 1;
          }
```

Reply:
It should compare categories first, and when equal compare shortcodes. Changed it to use String localeCompare.